src/Datadog.Trace/SpanContextPropagator.cs

```diff
@@ -91,22 +91,25 @@ private static ulong ParseUInt64(IHeadersCollection headers, string headerName)
         }
 
         private static T? ParseEnum<T>(IHeadersCollection headers, string headerName)
-            where T : struct
+            where T : struct, Enum
```

Comment:
huh, look at that, I wasn't aware you could constrain a type by Enum

Reply:
It was [added in C# 7.3](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/generics/constraints-on-type-parameters#enum-constraints).